lib/NetPacksLib.cpp

```diff
@@ -594,7 +594,11 @@ DLL_LINKAGE void HeroRecruited::applyGs( CGameState *gs )
 
 	h->setOwner(player);
 	h->pos = tile;
-	h->movement =  h->maxMovePoints(true);
+	bool fresh = (static_cast<int>(h->movement) == -1);

```

Comment:
See, there are so many places in VCMI where such magic numbers are used that fixing all of them is tons of work. Actually, I've got used to it. It is surely a good thing to make enums but I am fixing a bug in this PR, not trying to improve API in the first place.